src/app/features/ess-ui/ess-ui-exchangeset-request/ess-ui-exchangeset-request.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, OnInit, ViewChild, Input, ElementRef } from '@angular/core';
+import { FileInputComponent } from '@ukho/design-system';
+
+@Component({
+  selector: 'app-ess-ui-exchangeset-request',
+  templateUrl: './ess-ui-exchangeset-request.html',
+  styleUrls: ['./ess-ui-exchangeset-request.scss']
+})
+export class EssUiExchangesetRequestComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+  csvRecordsArray: any;
+  headersRow: any;
+  public records: any[] = [];
+  fileInputLabel = "ESS UI file upload for csv file";
+  @ViewChild('csvReader') csvReader: any;
+  jsondatadisplay: any;
+  messageType: 'info' | 'warning' | 'success' | 'error' = 'info';
+  messageDesc: string = "";
+  displayMessage: boolean = false;
+  @ViewChild("ukhoTarget") ukhoDialog: ElementRef;
+  errorMessageDescription = "";
+
+  uploadFileCsv($event: any): void {
+    this.records = [];
+    this.displayMessage = false;
+    let file;
+    if ($event.type == 'drop') {
+      file = $event.dataTransfer.files[0];
+    }
+    else {
+      file = $event.srcElement.files[0];
+    }
+    if (this.isValidCSVFile(file)) {
+      let reader = new FileReader();
+      reader.readAsText(file);
+      reader.onload = () => {
+        let csvData = reader.result;
+        this.csvRecordsArray = (<string>csvData).split(/\r\n|\n/);
+        this.csvRecordsArray = this.csvRecordsArray.map((str: string) => str.replace(/[�]| {2,}/g, ''));
+        this.headersRow = this.getHeaderArray(this.csvRecordsArray);
+        if (!this.validateCSVFile(this.csvRecordsArray, this.headersRow)) {
+          this.errorMessageDescription = this.errorMessageDescription;
+          this.showMessage(
+            "error",
+            this.errorMessageDescription);
+        }
+        else {
+          this.records = this.getDataRecordsFromCSVFile(
+            this.csvRecordsArray,
+            this.headersRow.length
+          );
+          this.hasRecordsToShow();
```

Comment:
`this.hasRecordsToShow();` does not do anything, please remove

Reply:
Hi @jimjamjahaa Thanks for the comment, I have removed this.hasRecordsToShow();. Please review.